routes/stats.go

```diff
@@ -9,8 +9,10 @@ import (
 
 func RegisterGetStats(r *mux.Router) {
 	r.HandleFunc("/student/exists/{username}", utils.JsonIO(controllers.CheckStudent)).Methods("GET")
+
 	r.HandleFunc("/students", utils.JsonIO(controllers.AllStudents)).Methods("GET")
-	r.HandleFunc("/student/{username}", utils.JsonIO(controllers.OneStudent)).Methods("GET")
+	r.HandleFunc("/student/{username}", utils.LoginRequired(utils.JsonIO(controllers.OneStudent))).Methods("POST")
```

Comment:
```suggestion
	r.HandleFunc("/student/{username}", utils.LoginRequired(utils.JsonIO(controllers.OneStudent))).Methods("GET")
```

shouldnt the endpoint be get , since it would be fetching data